V server readme (#1)
HerzogElias/vServer-Settup: README.md

# v-server-setup

> Readme Descritpion of V-Server-Setup Project. 

Reviewer: Specific Readme please. You do not set up your V-Server, but take care of the connection and configure nginx

This Project describes how do you can set up your own VServer. 

> ## Revequetits: 

Reviewer: ```suggestion
## Prerequisites: 
```

    You need a Ubuntu Cloud VM to setting your VServer.

> ## Qucikstart 

Reviewer: ```suggestion
## Quickstart
```

### Create SSH Keys 

### First Login with Passwort

### Deactivate Passwort Login 

### Install nginx 

#### Installing Process

> #### Create Alterantive Website for nginx

Reviewer: We dont need this here. You could add this to the Table of Contents. You can then click on the link and click on the relevant place in your readme, e.g.

## Table of Contents
1. [Introduction](#introduction)
1. [Description](#description)
1. [Quickstart](#quickstart)
1. [Usage](#usage)  
   - [Create SSH Keys](#create-ssh-keys)  
   - [First Login with Passwort](#first-login-with-passwort)  
   - [Deactivate Passwort Login](#deactivate-passwort-login)  
   - [Install nginx](#install-nginx)  
     - [Installing Process](#installing-process)  
     - [Create Alternative Website for nginx](#create-alternative-website-for-nginx)  
---
 ```## Table of Contents
1. [Introduction](#introduction)
1. [Description](#description)
1. [Quickstart](#quickstart)
1. [Usage](#usage)  
   - [Create SSH Keys](#create-ssh-keys)  
   - [First Login with Passwort](#first-login-with-passwort)  
   - [Deactivate Passwort Login](#deactivate-passwort-login)  
   - [Install nginx](#install-nginx)  
     - [Installing Process](#installing-process)  
     - [Create Alternative Website for nginx](#create-alternative-website-for-nginx)  
```

1. Create SSH-Keys with Sandard ed22591 on your local maschine. 
    ``` bash
    ssh-keygen -t ed25519 -C "deine-email@example.com"
    ```

2. Login to your VServer with your Username and passwort: 
    ```bash
    ssh <username>@<ip-adress> 
    ```

3. Use "type" on your local maschine to copy your generated public key on your VServer: 
    ``` bash
    type C:/Path/to/key| ssh username@ip-adress "cat >> /home/user/.ssh/authorized_keys"
    ```

4. Try to login to your VServer with your copied public key.
    ```bash
    ssh -i C:/Path1/to/key <usernema>@<ip-adress>
    ```

> 5. Deatkivate Password Login. 

Reviewer: ```suggestion
5. Deactivate Password Login. 
```

    1. Go to "etc/ssh/sshd.config" with Bash 
    ``` bash
    sudo nano /etc/ssh/sshd_config
    ```
    2. Serarch to "#PasswortAuthentication yes" and change it to "PasswortAuthentication no"
    3. Save the file and exit 
    4. Restart the sshd service. 
    ``` bash

>     sudo sysrtemctl restart ssh.service" 

Reviewer: ```suggestion
    sudo systemctl restart ssh.service" 
```

    ```

6. Update Respoitories on your Server. 
    ``` bash

>     sudo apt update" 

Reviewer: ```suggestion
    sudo apt update
```

    ```

7. Install Nginx on your Server.
    Use the following command: 
    ``` bash 
    sudo apt install nginx -y
    ```

8. Crearte Alternative html side for your nginx Serrver: 
    1. Create a new Directory 
    ``` bash
    mkdir -p /var/www/alternatives
    ```
    2. Create a new HTML file:
    ``` bash 
    sudo touch /var/www/alternatives/alternate-index.html
     ```
 
    3. Add a new configuration to: 
    ``` bash
    sudo nano /etc/nginx/sites-enabled/alternatives

>     ```

Reviewer: You can summarise this. With `nano` you create the file and can edit it directly.

# v-server-setup

## Table of Contents

1. [Introduction](#Introduction)
2. [Prerequisites](#Prerequisites)
3. [Usage](#Usage)  

>    - [Create SSH Keys](#Crearte-SSH-Keyws)  

Reviewer: ```suggestion
   - [Create SSH Keys](#Create-SSH-Keys)  
```

>    - [First Login with Passwort](#Login-wiith-Passwort)  

Reviewer: ```suggestion
   - [First Login with Passwort](#Login-with-Passwort)  
```

>    - [Deactivate Passwort Login](#Deactivae-Passwortlogin)  

Reviewer: ```suggestion
   - [Deactivate Passwort Login](#Deactivate-Passwortlogin)  
```

   - [Install nginx](#Install-Nginx)  
     - [Installing Process](#Installing-Process)  

>      - [Create Alternative Website for nginx](#Create-alternatvie-Webseite)  

Reviewer: ```suggestion
     - [Create Alternative Website for nginx](#Create-alternative-Website)  
```

    ```

5. Deatkivate Password Login. 
    1. Go to "etc/ssh/sshd.config" with Bash 
    ``` bash
    sudo nano /etc/ssh/sshd_config
    ```
    2. Serarch to "#PasswortAuthentication yes" and change it to "PasswortAuthentication no"
    3. Save the file and exit 
    4. Restart the sshd service. 
    ``` bash
    sudo sysrtemctl restart ssh.service" 
    ```
### Install Nginx
#### Insalling--Process 

6. Update Respoitories on your Server. 
    ``` bash
    sudo apt update" 
    ```

7. Install Nginx on your Server.
    Use the following command: 
    ``` bash 
    sudo apt install nginx -y
    ```
#### Create-alternatvie-Webseite
8. Crearte Alternative html side for your nginx Serrver: 
    1. Create a new Directory 
    ``` bash
    mkdir -p /var/www/alternatives
    ```
    2. Create a new HTML file:
    ``` bash 
    sudo touch /var/www/alternatives/alternate-index.html
     ```
 
    3. Add a new configuration to: 
    ``` bash
    sudo nano /etc/nginx/sites-enabled/alternatives
    ```
        1. Open File
        2. Add: 
            ``` bash
            server {
            listen 8081;
            listen [::]:8081;

            root /var/www/alternatives;

>             index alternate-index.html;git 

Reviewer: ```suggestion
            index alternate-index.html;
```

            location / {
                try_files $uri $uri/ =404;
            }
            ```
        3. Save this file and close it. 
    4. Open the new html file.
    ``` bash 
    sudo nano var/www/alternatives/alternate-index.html
    ```
    5. Take your HTML Code into this file. 
    6. Save and Close this file. 
    7. Open your new Websiete on "<your_ip>:8081"    

>     Link zum dso-github: https://github.com/Developer-Akademie-GmbH/dso-python-tasks/pull/5

Reviewer: What do you want to achieve with this? This is a pull request from your repo to the DA repo. That wasn't its purpose, was it? :)

> # v-server-setup

Reviewer: Please add a specific headline without `-`

> Readme Descritpion of V-Server-Setup Project 

Reviewer: Here is a description of what the repo does and what your goal is

1. Login to your VServer with your Username and passwort: 

>     ssh username@ip-adress 

Reviewer: We can better describe such commands in a repo, for example:

```suggestion
    ssh <your_username>@<your_ip>
```

But here the order is wrong: You want to log in with an SSH key, but you only create it in the next steps.

Reviewer: `ssh <your_username>@<your_ip>`

Reviewer: ```
ssh <your_username>@<your_ip>
ssh <your_username>@<your_ip>
```

2. if there are function of login with Username and passwort, logout from your VServer. 

3. Create SSH-Keys with Sandard ed22591 on your local maschine. 
    Use: ssh-keygen -t ed25519 -C "deine-email@example.com"

4. Use "type" on your local maschine to copy your generated public key on your VServer: 

>     Use: type C:/Path/to/key| ssh username@ip-adress "cat >> /home/user/.ssh/authorized_keys"

Reviewer: ```suggestion
    Use: type C:/Path/to/key| ssh <your_username>@<your_vm_ip> "cat >> /home/user/.ssh/authorized_keys"
```

> 5. Try to login to your VServer with your copied public key. The system dont ask you for a passwort. 

Reviewer: Too much information for a technical readme :)

```suggestion
5. Try to login to your VServer with your copied public key.
```

    ssh -i C:/Path1/to/key usernema@ipadress

6. Logout from your VServer and repeat Step 5. 

>     Make sure that you can login with your SSH Key! 

Reviewer: We don't need this step

7. Deatkivate Password Login. 
    1. Go to "etc/ssh/sshd.config" with Bash "sudo nano /etc/ssh/sshd_config"
    2. Serarch to "#PasswortAuthentication yes" and change it to "PasswortAuthentication no"
    3. Save the file and exit 
    4. Restart the sshd service with "sudo sysrtemctl restart ssh.service" 

8. Logut form Server 

9. Try Login with username and passwort like at stap 1. You must get a "Percession denied" 

> 10. Login with SSH to your VServer. 

Reviewer: This is a description of how you did it. But as a user, I don't need this information to use your repo.

11. Update Respoitories on your Server. 
    Use: "sudo apt update" 

12. Install Nginx on your Server.
    Use: "sudo apt install nginx -y"

13. Crearte Alternative html side for your nginx Serrver: 

>     1. Create a new Directory with "mkdir /var/www/alternatvies" 

Reviewer: ```suggestion
    1. Create a new Directory with "mkdir -p /var/www/alternatives" 
```

`-p` ensures the directory is created if it doesn't exist.

    2. Create a new HTML file with "sudo touch /var/www/alternatives/alternate-index.html"
    3. Add a new configuration to: "sudo nano /etc/nginx/sites-enabled/alternatives"
        1. Open File
        2. Add: 
            server {
            listen 8081;
            listen [::]:8081;

            root /var/www/alternatives;
            index alternate-index.html;

            location / {
                try_files $uri $uri/ =404;
            }
        3. Save this file and close it. 
    4. Open the new html file with "sudo nano var/www/alternatives/alternate-index.html"
    5. Take your HTML Code into this file. 
    6. Save and Close this file. 

>     7. Open your new Websiete on "your-ip-adress:8081"    

Reviewer: ```suggestion
    7. Open your new Websiete on "<your_ip>:8081"    
```

>     Link zum dso-github: https://github.com/Developer-Akademie-GmbH/dso-python-tasks/compare/main...HerzogElias:dso-python-tasks:module-2?expand=1

Reviewer: Here the mistake has happened again: This is the link to create the pull request in our DSO Python repo. What else would you like to give us?
